Lombiq.Tests.UI/Extensions/HttpClientUITestContextExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Lombiq.Tests.UI.Services;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lombiq.Tests.UI.Extensions;
+
+[SuppressMessage(
+    "Security",
+    "SCS0004: Certificate Validation has been disabled.",
+    Justification = "Certificate validation is unnecessary for UI testing.")]
+[SuppressMessage(
+        "Reliability",
+        "CA2000:Dispose objects before losing scope",
+        Justification = "Disposed by the HttpClient.")]
+public static class HttpClientUITestContextExtensions
+{
+    /// <summary>
+    /// Creates a new <see cref="HttpClient"/> and authorizes it with a Bearer token that is created based on the provided
+    /// <paramref name="clientId"/> and <paramref name="clientSecret"/>.
+    /// </summary>
+    public static async Task<HttpClient> CreateAndAuthorizeClientAsync(
```

Comment:
Why don't you use the https://github.com/Lombiq/Orchard-Core-API-Client library of ours?

Reply:
Seemed simpler this way with no drawbacks. What would be the advantages of using the library?

Reply:
I think you are reusing fairly big part of it already by copying code from it. Isn't it possible to DRY this with that and abstracting the already existing logic and doing the same there?

Reply:
I only took the `HttpClientHandler` settings and its suppression messages from the library.

The issue I have with using the library is it also entails adding new interface(s) for our app's endpoints, if I understand correctly (just like the existing ones for e.g. [tenant-related operations](https://github.com/Lombiq/Orchard-Core-API-Client/blob/dev/Lombiq.OrchardCoreApiClient/Interfaces/IOrchardCoreTenantsApi.cs)). While I think we could use the ApiClient class instead of the generic HttpClient one to issue requests, I don't see the upside of making it work, especially as the solution I added is even more simple and doesn't require more submodules.

It's possible that I'm also biased here and not entirely objective (frankly, this task has become very tedious), so if you definitely think using the library would be a better approach, I'll do it, I just haven't found good enough reason to do so so far.

Reply:
You don't necessarily need to create interfaces. You can create a new class instead like the ApiClient, but it would be for your usecase. But you also need to make sure about disposing the HTTP client like it is done in ApiClient. Because it seems like the memory usage is quite high in OFFI compared to OSOCE where we have 4x more UI tests, so that's weird.

Reply:
Or if that just doesn't seem worth it, make sure you dispose the httpclient always.

Reply:
Ok, revert disposing.